src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenRefLocalTests.cs

```diff
@@ -3710,5 +3710,157 @@ void verify(CSharpCompilationOptions options, Verification verify, string expect
                 verifier.VerifyIL("<top-level-statements-entry-point>", expectedIL);
             }
         }
+
+        [Fact]
+        [WorkItem(60905, "https://github.com/dotnet/roslyn/issues/60905")]
+        public void ReadValueAndDiscard_01()
+        {
+            var source =
+@"struct S { }
+class Program
+{
+    static void Main()
+    {
+        F(new S[1]);
+    }
+    static void F(S[] a)
+    {
+        ref var b = ref a[0];
+        _ = b;
+    }
+}";
+            var verifier = CompileAndVerify(source, options: TestOptions.DebugExe, expectedOutput: "");
+            verifier.VerifyIL("Program.F",
+@"{
+  // Code size       11 (0xb)
+  .maxstack  2
+  .locals init (S& V_0) //b
+  IL_0000:  nop
+  IL_0001:  ldarg.0
+  IL_0002:  ldc.i4.0
+  IL_0003:  ldelema    ""S""
+  IL_0008:  stloc.0
+  IL_0009:  nop
+  IL_000a:  ret
+}");
+            verifier = CompileAndVerify(source, options: TestOptions.ReleaseExe, expectedOutput: "");
+            verifier.VerifyIL("Program.F",
+@"{
+  // Code size        9 (0x9)
+  .maxstack  2
+  IL_0000:  ldarg.0
+  IL_0001:  ldc.i4.0
+  IL_0002:  ldelema    ""S""
+  IL_0007:  pop
+  IL_0008:  ret
+}");
+        }
+
+        [Fact]
+        public void ReadValueAndDiscard_02()
+        {
+            var source =
+@"struct S { }
+class Program
+{
+    static void Main()
+    {
+        var s = new S();
+        F(ref s);
+    }
+    static void F(ref S s)
+    {
+        _ = s;
```

Comment:
Should `_ = s;` result in a `NullReferenceException` at runtime if the reference is `null`?

Reply:
From discussion with @jaredpar, we should emit the dereference of `ref s` here, even though the value is discarded.

Reply:
My logic here was essentially that it's possible for a `ref` to be `null` hence this is no different than say `_ = a.b` where `b` is a field / prop and `a` is a reference type. It is an operation that has a potential visible side effect of throwing a `NullReferenceException`. 

Before c# 11 there was no supported way in C# to create a `ref` that points to `null` which meant this was probably a lower priority issue. I would've felt a bit more ambivalent about it. Now that we can create such `ref` I feel a bit more confident that we should be treating it as a side effect here.

Reply:
Let's emit the dereference as part of the ref fields feature. I've added an item to the [ref fields test plan](https://github.com/dotnet/roslyn/issues/59194).